C++/Algorithms/Mathematical/SquareRoot.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+using namespace std;
+
+
+//These 2 functions will accept user input and return the resulting Square Root of that user input
+
+
+//Iterative Function
+double sqrtIterative(double x) {
+	double sqrt = x * pow(0.3, log10(x)); //Square Root Value
+	for (int i = 0; i < 5; i++)
+		sqrt = (sqrt + (x / sqrt)) * 0.5;
+	return sqrt;
+}
+
+
+//Recursive function
+double sqrtRecursive(double num, double prev)
+{
+	double next = (prev + num / prev) / 2;  //Smallest Square Root Value
+	if (fabs(next - prev) < DBL_EPSILON * next)
+		return next;
+	return sqrtRecursive(num, next);
+}
+
+int main(void) {
+
+
+
+	//Iterative Function
+
+	cout << "Iterative Function" << endl;
+	cout << "Please enter a number" << endl;
+	double iFn;
+	cin >> iFn;
+	double answerIterative = sqrtIterative(iFn);
+	cout << answerIterative << endl;
+
+
+	//Recursive Function
+	cout << "Recursive Function" << endl;
+	cout << "Please enter 2 numbers, the first will be the square root number" << endl;
```

Comment:
To demonstrate the recursive algorithm, instead of asking the user for a second number, you could define another constant or even generate a random number within a range specified and pass it as an argument to your ``sqrtRecursive()`` function thereby asking the user only to specify a single number that which they desire to find the square root of.